pkg/cloudprovider/provider/anexia/provider.go

```diff
@@ -270,7 +272,7 @@ func ensureConditions(status *anxtypes.ProviderStatus) {
 	}
 }
 
-func (p *provider) resolveConfig(config anxtypes.RawConfig) (*resolvedConfig, error) {
+func (p *provider) resolveConfig(ctx context.Context, config anxtypes.RawConfig, retrieveGenericClient func(token string) (api.API, error)) (*resolvedConfig, error) {
```

Comment:
This generic client retriever was added for having a mock client in testing, right? Why is the generic client not passed as argument to this function directly? Because we need the token, which is resolved in this function? We could split the template resolve stuff in its own function, passing that the client to use and test that function on its own with a mock client we just give it

Reply:
> Because we need the token, which is resolved in this function?

Correct.

> We could split the template resolve stuff in its own function, passing that the client to use and test that function on its own with a mock client we just give it

Just to clarify, because I think I had it implemented this way before: `resolveConfig` would return `resolvedConfig` without the `TemplateID` set and we'd then fill in the `TemplateID` afterwards, correct?

Reply:
Wouldn't this break testability of the `resolveConfig` method as we can no longer inject the mock API client?

Reply:
if `resolveConfig` only calls `resolveTemplate` when `TemplateID` isn't set, we can just have a `TemplateID` for all `resolveConfig` test cases